Microsoft.Toolkit.Uwp.UI.Controls.Input/TokenizingTextBox/TokenizingTextBox.Properties.cs

```diff
@@ -157,6 +157,46 @@ private static void TextPropertyChanged(DependencyObject d, DependencyPropertyCh
             typeof(TokenizingTextBox),
             new PropertyMetadata(false));
 
+        /// <summary>
+        /// Identifies the <see cref="MaxTokens"/> property.
+        /// </summary>
+        public static readonly DependencyProperty MaxTokensProperty = DependencyProperty.Register(
+            nameof(MaxTokens),
+            typeof(int),
+            typeof(TokenizingTextBox),
+            new PropertyMetadata(null, OnMaxTokensChanged));
+
+        private static void OnMaxTokensChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            if (d is TokenizingTextBox ttb && ttb.ReadLocalValue(MaxTokensProperty) != DependencyProperty.UnsetValue && e.NewValue is int newMaxTokens)
+            {
+                var tokenCount = ttb.Items.Count;
+                if (tokenCount > newMaxTokens)
```

Comment:
When testing out this event, I always seem to get `tokenCount` of 1 (an empty string item), no matter how many valid tokens existed in the corresponding TTB before modifying MaxTokens

Reply:
Also related to the sample refresh thing, when the value changes, this code doesn't actually get run beyond the first time it gets set. In which case there is always 1 input textbox item in the list by default, hence the `tokenCount` of 1. 

All that said, this method is fairly inefficient, because I'm playing around with the `ttb._innerItemsSource` directly (which has all the items, interspersed and tokens) instead of working with the underlying `ttb._innerItemsSource.ItemsSource` (which has only the tokens). I was able to simplify the logic a fair bit 👍